crates/forge/bin/cmd/build.rs

```diff
@@ -87,35 +87,36 @@ impl BuildArgs {
             project = config.project()?;
         }
 
-        let mut compiler = ProjectCompiler::new()
-            .print_names(self.names)
-            .print_sizes(self.sizes)
-            .quiet(self.format_json)
-            .bail(!self.format_json);
-        if let Some(skip) = self.skip {
-            if !skip.is_empty() {
-                compiler = compiler.filter(Box::new(SkipBuildFilters::new(skip)?));
+        if !config.zksync {
+            let mut compiler = ProjectCompiler::new()
+                .print_names(self.names)
+                .print_sizes(self.sizes)
+                .quiet(self.format_json)
+                .bail(!self.format_json);
+            if let Some(skip) = self.skip {
+                if !skip.is_empty() {
+                    compiler = compiler.filter(Box::new(SkipBuildFilters::new(skip)?));
+                }
             }
-        }
-        let output = compiler.compile(&project)?;
 
-        if self.format_json {
-            println!("{}", serde_json::to_string_pretty(&output.clone().output())?);
-        }
-
-        if config.zksync {
+            let output = compiler.compile(&project)?;
+            if self.format_json {
+                println!("{}", serde_json::to_string_pretty(&output.output())?);
+            }
+        } else {
             let zk_compiler = ProjectCompiler::new()
                 .print_names(self.names)
                 .print_sizes(self.sizes)
                 .quiet(self.format_json)
                 .bail(!self.format_json);
-            let zk_output = zk_compiler.zksync_compile(&project)?;
+
+            let output = zk_compiler.zksync_compile(&project)?;
```

Comment:
I would maintain zk_output to be clear which one is it faster but given it will only run one branchf of the condition I guess its fine!